journal.py

```diff
@@ -86,22 +94,41 @@ def home(request):
 
 @view_config(route_name='add', renderer='templates/add.jinja2')
 def add(request):
-    entries = Entry.all()
-    return {'entries': entries}
+    if request.method == 'POST':
+        title = request.params.get('title')
+        text = request.params.get('text')
+        if not (title == "" or text == ""):
+            Entry.write(title=title, text=text)
+            return HTTPFound(request.route_url('home'))
+        else:
+            return {'title': title, 'text': text}
+    else:
+        return {'title': '', 'text': ''}
 
 
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail(request):
-    entries = Entry.all()
-    return {'entries': entries}
+    entry_id = request.matchdict['entry_id']
+    title, text, time = Entry.get_entry(entry_id)
+    text = markdown.markdown(text, extensions=['codehilite', 'fenced_code'])
+    return {'title': title, 'text': text, 'id': entry_id, 'time': time}
 
 
-@view_config(route_name='add', request_method='POST')
-def add_entry(request):
-    title = request.params.get('title')
-    text = request.params.get('text')
-    Entry.write(title=title, text=text)
-    return HTTPFound(request.route_url('home'))
+@view_config(route_name='edit', renderer='templates/edit.jinja2')
+def edit(request):
+    entry_id = request.matchdict['entry_id']
+    if request.method == 'GET':
+        title, text, time = Entry.get_entry(entry_id)
+        return {'title': title, 'text': text, 'id': entry_id}
+
+    if request.method == 'POST':
+        title = request.params.get('title')
+        text = request.params.get('text')
+        if not (title == "" or text == ""):
+            Entry.edit_entry(entry_id, title, text)
+            return HTTPFound(request.route_url('home'))
+        else:
+            return {'title': title, 'text': text, 'id': entry_id}
```

Comment:
I count three possible return points from this function.  That's a lot.  So many ways out can be confusing to read.